File: Lista de Exercícios/Go/9. Interfaces & Protocolos/Interface.go
package main

import (
	"fmt"
)

type Imprimivel interface {
	imprimir()
}

type Relatorio struct{}

func (r Relatorio) imprimir() {
	fmt.Println("O relatório está em processo de impressão")
}

type Contrato struct{}

func (c Contrato) imprimir() {
	fmt.Println("O contrato está em processo de impressão")
}

func main() {

	var documentos []Imprimivel
	documentos = append(documentos, Relatorio{}, Contrato{})

	for _, documento := range documentos {
		documento.imprimir()
	}
}

File: Lista de Exercícios/Java/9. Interfaces & Protocolos/Interface.java
interface Imprimivel {
    void imprimir();
}

class Relatorio implements Imprimivel {
    @Override
    public void imprimir() {
        System.out.println("O relatório está em processo de impressão");
    }
}

class Contrato implements Imprimivel {
    @Override
    public void imprimir() {
        System.out.println("O contrato está em processo de impressão");
    }
}